File: hw-3B/C++/B-seen-numbers.cpp
#include <iostream>
#include <vector>
#include <fstream>
#include <unordered_map>

using namespace std;

void isSeenBefore(vector <int> arr){

    unordered_map <int, int> dct;
    for (auto item: arr){
        if (!dct.count(item)){
            dct[item] = 0;
            cout << "NO" << '\n';
        }
        dct[item] ++;
        if (dct[item] > 1)
            cout << "YES" << '\n';
    }
}

int main(){

    // Print whether the number is seen before

    ifstream fin("input.txt");
    int x;
    vector <int> nums;
    while (fin >> x) {
        nums.push_back(x);
    }
    fin.close();

    isSeenBefore(nums);

    return 0;
}

File: hw-3B/Python/B-seen-numbers.py
def is_seen_before(arr):

    dct = {}
    for item in arr:
        if item not in dct:
            dct[item] = 0
            print('NO')
        dct[item] += 1
        if dct[item] > 1:
            print('YES')


if __name__ == '__main__':

    """Print whether the number is seen before"""

    line1 = [int(x) for x in input().split()]
    is_seen_before(line1)
